### ISSUE TITLE
Epic: complete pageserver `import_pgdata` API

### ISSUE THREAD

**Issue by jcsp** [neondatabase/neon#10188]
The pageserver has an experimental `import_pgdata` attribute in the timeline creation API, which can be used for ingesting data from native postgres format as a bulk operation.  This ticket tracks work to complete this API to the point that it's ready for use in production.

### Required
- [x] https://github.com/neondatabase/cloud/issues/21876
- [x] https://github.com/neondatabase/cloud/issues/22026
- [x] https://github.com/neondatabase/cloud/issues/21350
- [x] https://github.com/neondatabase/neon/issues/11566
- [x] https://github.com/neondatabase/neon/issues/11570
- [x] https://github.com/neondatabase/neon/issues/11567
- [x] https://github.com/neondatabase/neon/issues/11568
- [x] https://github.com/neondatabase/neon/issues/11569
- [x] https://github.com/neondatabase/neon/issues/11664
- [x] https://github.com/neondatabase/neon/issues/11884
- [x] https://github.com/neondatabase/neon/issues/11983
- [x] https://github.com/neondatabase/neon/issues/11897
- [X] Address `TODO` comments in `flow.rs`
- [X] Address locations where we do potentially unbounded allocations in `flow.rs`
- [x] https://github.com/neondatabase/neon/issues/10191
- [x] https://github.com/neondatabase/neon/issues/11800

### Nice to have
- [ ] Revise API (discovering status of long running operations, etc)
- [x] https://github.com/neondatabase/cloud/issues/28205
- [ ] https://github.com/neondatabase/cloud/issues/20646
- [x] https://github.com/neondatabase/neon/issues/10996

**Comment by VladLazar** [comment]
Status as of 2025-06-09

Storage side of fast-import is complete apart from the initial shard split before starting the import ([issue](https://github.com/neondatabase/neon/issues/11800)).

Next steps:
* wait for the import job to support larger workloads
* test it out with a TiB sized database
* write up some instructions and deliver it to the support team as an internal tool for migrating large databases
